### ISSUE TITLE
Provide checks to use pluralized variables names when accessing collections

### ISSUE THREAD

**Issue by andrueastman** [microsoftgraph/microsoft-graph-explorer-api#40]
Looking at this sample
```cs
GraphServiceClient graphClient = new GraphServiceClient( authProvider );
var user = await graphClient.Users
    .Request()
    .Select( e => new {
             e.DisplayName,
             e.GivenName,
             e.PostalCode 
             })
    .GetAsync();
```

The variable name used is in singular but we are actually returning a collection of users. 

We can use this library https://www.nuget.org/packages/Pluralize.NET.Core/ to pluralize variable names when we need to.

**Comment by MIchaelMainer** [comment]
Are you able to determine whether the query targets an entityset? If so, I think we want to just use the name of the entityset for the variable name. 

        <EntitySet Name="organization" EntityType="microsoft.graph.organization" />
        <EntitySet Name="groupSettings" EntityType="microsoft.graph.groupSetting" />
        <EntitySet Name="groupSettingTemplates" EntityType="microsoft.graph.groupSettingTemplate" />
        <EntitySet Name="subscribedSkus" EntityType="microsoft.graph.subscribedSku" />
        <EntitySet Name="users" EntityType="microsoft.graph.user">

var users = await graphClient.**Users**
var organization = await graphClient.**Organization**

In the case of the navigation, we just use the name of navigation property.

Eventually, we should show the reader the type of object returned. Let's not address this now.
UserCollectionPage users = await graphClient.Users

**Comment by andrueastman** [comment]
Closed with PR #41
